### PR TITLE
02 core get /api/topics

### PR DESCRIPTION
**Pull request by RKHarrison**
Hi again NC team!

I have built the GET /api/topics endpoint
-adds happy path and testing
-adds middleware to handle 500 server errors in-house
-adds generic 404 error handling and testing for requests made to any undefined endpoint

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+});
+
+describe("GET /api/topics", () => {
+  it("200: responds with an array of objects with correct properties", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const topics = body.topics;
+        expect(topics).toHaveLength(3);
```
Comment (MRombi) [inline __tests__/app.test.js:21]: Well done for checking the length here to avoid false positives 🙂

File: api/models/app.topics.models.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const db = require("../../db/connection")
+
+exports.fetchTopics = () => {
+    console.log('fetching');
```
Comment (MRombi) [inline api/models/app.topics.models.js:4]: We should remove console logs

**Review by MRombi — COMMENTED** [review]
Great job overall! 😄

**Comment by RKHarrison** [comment]
Thanks Michael, is there anything you think I could improve on?

**Comment by RKHarrison** [comment]
PS I saw the log as soon as i sent the pr 😅

**Pull request merged**